src/test/javascript/spec/component/programming-exercise/programming-exercise.component.spec.ts

```diff
@@ -70,6 +88,139 @@ describe('ProgrammingExercise Management Component', () => {
         // THEN
         expect(service.findAllProgrammingExercisesForCourse).toHaveBeenCalled();
         expect(comp.programmingExercises[0]).toEqual(expect.objectContaining({ id: programmingExercise.id }));
+        expect(comp.filteredProgrammingExercises[0]).toEqual(expect.objectContaining({ id: programmingExercise.id }));
+    });
+
+    it('Should reset exercise', () => {
+        const headers = new HttpHeaders().append('link', 'link;link');
+        jest.spyOn(exerciseService, 'reset').mockReturnValue(
+            of(
+                new HttpResponse({
+                    body: undefined,
+                    headers,
+                }),
+            ),
+        );
+        const mockSubscriber = jest.fn();
+        comp.dialogError$.subscribe(mockSubscriber);
+
+        comp.course = course;
+        comp.ngOnInit();
+        comp.resetProgrammingExercise(456);
+        expect(exerciseService.reset).toHaveBeenCalledWith(456);
+        expect(mockSubscriber).toHaveBeenCalledWith('');
+    });
+
+    it('Should not reset exercise', () => {
```

Comment:
Please add here the circumstance under which the exercise should not be reset or else it doesn't convey any information.

Please also adapt this for other occurrences in this or other files you changed.